core/src/main/scala/org/apache/spark/HeartbeatReceiver.scala

```diff
@@ -79,6 +79,17 @@ private[spark] class HeartbeatReceiver(sc: SparkContext, clock: Clock)
 
   private val checkTimeoutIntervalMs = sc.conf.get(Network.NETWORK_TIMEOUT_INTERVAL)
 
+  private val executorHeartbeatIntervalMs = sc.conf.get(config.EXECUTOR_HEARTBEAT_INTERVAL)
+
+  require(checkTimeoutIntervalMs < executorTimeoutMs,
```

Comment:
Last question: should this be <= ? that seems like the natural maximum. The message would need to be updated slightly too. I think you don't need to reference checkTimeoutIntervalMs etc in the message as they are variable names. Just refer to the .key strings directly.

Reply:
Anyway, I will update as your suggestion for I can't find a better way to make this log more clear.